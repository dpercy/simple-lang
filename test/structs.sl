(struct (Derp))
(Derp)
Derp
(def d Derp)
(d)
d
(match (Derp)
  [(d) "renamed constructors are patterns too"]
  [otherwise "uh oh"])
(def (f)
  derp)
(def p (f))
(match (Derp)
  [(p) "computed constructors are patterns too"]
  [otherwise "hm"])
(Derp 999)
(struct (Triple x y z))
Triple
(Triple)
(Triple "a" 3 (Derp))
(Triple d f Triple)
(match (Triple 1 2 3)
  [(Triple) "bad"]
  [(Triple x y z) "good"])
