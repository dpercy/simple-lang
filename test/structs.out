(Derp)
Derp
(Derp)
Derp
"renamed constructors are patterns too"
"computed constructors are patterns too"
(error "Derp: expected 0 arguments, but got 1")
Triple
(error "Triple: expected 3 arguments, but got 0")
(Triple "a" 3 (Derp))
(Triple Derp f Triple)
(error "Triple: has 3 fields, but pattern has 0")
